### PR TITLE
Improve shrike documentation (first round)

### PR DESCRIPTION
**Pull request by haozhestat**
Fix some typos and mistakes in the shrike documentation, and refactor some pages to align with those in the aka.ms/aml/compliant.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ please see the [docs page](http://azure.github.io/shrike).
 
 ## Installation
 
-The library `shrike` is publicly available in PyPi. There are three optional extra dependencies - `pipeline`, `build` and `dev`, 
+The `shrike` library is publicly available in the PyPi. There are three optional extra dependencies - `pipeline`, `build` and `dev`,
```
Comment (dkmiller) [inline README.md:38]: Just "in PyPI".

**Review by dkmiller — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (haozhestat) [reply README.md]: Addressed.

**Review by dkmiller — APPROVED** [review]

File: docs/build/build-for-sign-and-register.md [hunk 2]
```diff
@@ -16,7 +16,8 @@ To enjoy this tutorial, you need to
 
 
 ## Configuration
-Command line arguments and configuration YAML file are both supported by `shrike.build`. The order of precedence from least to greatest (the last listed variables override all other variables) is: default values, configuration file, command line arguments.
+Command line arguments and configuration YAML file are both supported by `shrike` library. Their priority is, in order from lowest to highest, default value < configuration file < command line arguments. Command line arguments have the highest priority, overriding all default
```
Comment (fuhuifang) [inline docs/build/build-for-sign-and-register.md:19]: I think we had a discussion before - this change seems making the sentences redundant. Just my opinion, you don't have to take it 😄

File: docs/build/build-for-sign-and-register.md [hunk 3]
```diff
@@ -57,34 +54,58 @@ To consume this configuration file, we should pass its path to the command line,
...
+successfully merged, but with the build failing. In these cases, the new
+components introduced/modified in the problematic PR have not been
+signed/registered, and unless they are modified by a subsequent PR, they will
+not be picked up by the "smart" mode. There are 2 common workarounds to this
+issue. The most straightforward is to activate the "all" activation mode in a PR
+following the failed build, then revert to "smart" for the PR after that. This
+will ensure all components are registered, but will also mess up the 
+[components results recycling logic](./Component-results-recycling.md):
+some components will wrongly be considered as a new, hence their results won't
+be recycled. The second option is to do a mock PR that just adds a commment to
```
Comment (fuhuifang) [inline docs/build/build-for-sign-and-register.md:95]: or bump up the version if they do not have any meaningful comments in mind?

**Review by fuhuifang — APPROVED** [review]

File: docs/build/build-for-sign-and-register.md (hunk 3 above)
Comment (haozhestat) [reply docs/build/build-for-sign-and-register.md]: Good comment. Addressed.

**Review by fuhuifang — APPROVED** [review]

**Pull request merged**